test/Microsoft.AspNet.Mvc.Core.Test/SessionStateTempDataProviderTest.cs

```diff
@@ -76,6 +76,44 @@ public void Save_NullSession_NonEmptyDictionary_Throws()
             });
         }
 
+        [Theory]
+        [MemberData(nameof(InvalidTypes))]
+        public void Save_InvalidType_Throws(string key, object value, Type type)
```

Comment:
Can you add test to verify that all the types mentioned in `TypeHelper.IsSimpleType` can be deserialized...I am specifically interested about `struct` and also not sure about `Uri`

https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Common/TypeHelper.cs#L68